src/spikeinterface/sortingcomponents/matching/wobble.py

```diff
@@ -0,0 +1,929 @@
+
+        # We want to enforce refractory conditions on unit_indices rather than template_indices for units with many templates
+        spike_unit_indices = spike_template_indices.copy()
+        for template_id in set(spike_template_indices):
+            unit_id = template_meta.template_indices2unit_indices[template_id] # unit_id corresponding to this template
+            spike_unit_indices[spike_template_indices==template_id] = unit_id
+
+        # Get the samples (time indices) that correspond to the waveform for each spike
+        waveform_samples = get_convolution_len(spike_times[:, np.newaxis], template_meta.num_samples) + window
+
+        # Enforce refractory by setting objective to negative infinity in invalid regions
+        objective_normalized[spike_unit_indices[:, np.newaxis], waveform_samples[:, 1:-1]] = -1 * np.inf
+        if params.scale_amplitudes: # template_convolution is only used with amplitude scaling
+            objective[spike_unit_indices[:, np.newaxis], waveform_samples[:, 1:-1]] = -1 * np.inf
+
+
+def compute_template_norm(visible_channels, templates):
+    """Computes norm of each template.
+
+    Parameters
+    ----------
+    visible_channels : ndarray (num_units, num_channels)
+        visible_channels[unit, channel] is True if the unit's template has sufficient amplitude on that channel.
+    templates : ndarray (num_templates, num_samples, num_channels)
+        Spike template waveforms.
+
+    Returns
+    -------
+    norm : ndarray (num_templates,)
+    Magnitude of each template for normalization.
+    """
+    num_templates = templates.shape[0]
+    norm = np.zeros(num_templates, dtype=np.float32)
+    for i in range(num_templates):
+        norm[i] = np.sum(
+            np.square(templates[i, :, visible_channels[i, :]])
+        )
+    return norm
+
+
+def compress_templates(templates, params, template_meta):
+    """Compress templates using singular value decomposition.
+
+    Parameters
+    ----------
+    templates : ndarray (num_templates, num_samples, num_channels)
+        Spike template waveforms.
+    params : WobbleParameters
+        Parameters for WobbleMatch algorithm.
+    template_meta : TemplateMetadata
+        Dataclass object for aggregating template metadata together.
+
+    Returns
+    -------
+    compressed_templates : (ndarray, ndarray, ndarray, ndarray)
+        Templates compressed by singular value decomposition into temporal, singular, spatial, and
+        upsampled_temporal components.
+    """
+    temporal, singular, spatial = np.linalg.svd(templates)
+
+    # Keep only the strongest components
+    temporal = temporal[:, :, :params.conv_approx_rank]
+    singular = singular[:, :params.conv_approx_rank]
+    spatial = spatial[:, :params.conv_approx_rank, :]
+
+    # Upsample the temporal components of the SVD -- i.e. upsample the reconstruction
+    if params.jitter_factor == 1:  # Trivial Case
+        temporal = np.flip(temporal, axis=1)
```

Comment:
I see. I was -wrongly- assuming that the rightmost matrix in the SVD decomposition should be conjugate, but now I see that numpy's algorithm is already returning the conjugate. 

I would like to suggest changing the initialization reverse with filp to view version that operates in place (that is, use the reversed view in the `compute_objective` and `convolve_templates` functions). Using the reversed sliced notation (see screenshoot) avoids creating and internal copy and can be done in the specific lines where that is required without extra cost. This should not only be faster but also will happen to avoid reading mistakes (i.e. somebody forgetting that the temporal component is reversed in time).

![image](https://user-images.githubusercontent.com/6429509/231083555-0e26e586-68ee-4269-b026-9e5d5c3fcbf6.png)